File: doc/src/scripts/client/csharp/CollectionEvent.cs
using System;
using System.Collections.Generic;
using KRPC.Client;
using KRPC.Client.Services.KRPC;
using Type = KRPC.Client.Services.KRPC.Type;
using KRPC.Client.Services.SpaceCenter;

class Program {
    public static void Main () {
        var connection = new Connection ();
        var krpc = connection.KRPC ();
        var vessel = connection.SpaceCenter ().ActiveVessel;

        // A function taking an engine, returning true if it is out of fuel
        var engine = Expression.Parameter (connection, "engine",
            Type.ClassType (connection, "SpaceCenter", "Engine"));
        var hasFuel = Connection.GetCall (() => vessel.Parts.Engines [0].HasFuel);
        var outOfFuel = Expression.Lambda (connection,
            new List<Expression> { engine },
            Expression.Not (connection,
                Expression.CallWithArguments (connection, hasFuel,
                    new Dictionary<int, Expression> { { 0, engine } })));

        // Whether any engine satisfies it
        var engines = Connection.GetCall (() => vessel.Parts.Engines);
        var expr = Expression.Any (connection,
            Expression.Call (connection, engines), outOfFuel);

        var evnt = krpc.AddEvent (expr);
        lock (evnt.Condition) {
            evnt.Wait ();
            Console.WriteLine ("An engine has run out of fuel");
        }
    }
}

File: doc/src/scripts/client/java/CollectionEvent.java
import krpc.client.Connection;
import krpc.client.Event;
import krpc.client.RPCException;
import krpc.client.StreamException;
import krpc.client.services.KRPC;
import krpc.client.services.KRPC.Expression;
import krpc.client.services.KRPC.Type;
import krpc.client.services.SpaceCenter;
import krpc.schema.KRPC.ProcedureCall;

import java.io.IOException;
import java.util.Arrays;
import java.util.Collections;

public class CollectionEvent {
  public static void main(String[] args) throws IOException, RPCException, StreamException {
    try (Connection connection = Connection.newInstance()) {
      KRPC krpc = KRPC.newInstance(connection);
      SpaceCenter spaceCenter = SpaceCenter.newInstance(connection);
      SpaceCenter.Vessel vessel = spaceCenter.getActiveVessel();

      // A function taking an engine, returning true if it is out of fuel
      Expression engine = Expression.parameter(
        connection, "engine",
        Type.classType(connection, "SpaceCenter", "Engine"));
      ProcedureCall hasFuel = connection.getCall(
        vessel.getParts().getEngines().get(0), "getHasFuel");
      Expression outOfFuel = Expression.lambda(
        connection, Arrays.asList(engine),
        Expression.not(connection, Expression.callWithArguments(
          connection, hasFuel, Collections.singletonMap(0, engine))));

      // Whether any engine satisfies it
      ProcedureCall engines = connection.getCall(vessel.getParts(), "getEngines");
      Expression expr = Expression.any(
        connection, Expression.call(connection, engines), outOfFuel);

      Event event = krpc.addEvent(expr);
      synchronized (event.getCondition()) {
        event.waitFor();
        System.out.println("An engine has run out of fuel");
      }
    }
  }
}
